packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/assign-public-ip/assign-public-ip.ts

```diff
@@ -0,0 +1,69 @@
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as route53 from '@aws-cdk/aws-route53';
+import * as cdk from '@aws-cdk/core';
+import { Service } from '../../service';
+import { ServiceExtension, ServiceBuild, EnvironmentCapacityType } from '../extension-interfaces';
+import { TaskRecordManager } from './task-record-manager';
+
+export interface AssignPublicIpExtensionOptions {
+  /**
+   * DNS Zone to expose expose a record to. If specified, `dnsRecordName` is
```

Comment:
To make the optional/required behavior a bit clearer let's put both these properties into their own interface that is optional, but this sub interface has both properties required. That way the TypeScript compiler can catch misuse rather than needing a runtime check.

```ts
new AssignPublicIpExtension({
  dns: {
    dnsZone: zone,
    dnsRecordName: name
  }
})
```

`dns` is optional, but inside `dns` both subprops are required.

Reply:
Great idea! I'll make this fix asap